RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -118,63 +169,114 @@ class EvtPlaneProducer : public edm::EDProducer {
   virtual void endJob() ;
   
   // ----------member data ---------------------------
-  edm::EDGetTokenT<reco::VertexCollection> vtxCollection_;
-  edm::EDGetTokenT<CaloTowerCollection> caloCollection_;
-  edm::EDGetTokenT<reco::TrackCollection> trackCollection_;
+  // edm::InputTag vtxCollection_;
+  // edm::InputTag caloCollection_;
+  // edm::InputTag trackCollection_;
+
+  edm::Service<TFileService> fs;
+
+  std::string centralityVariable_;
+  std::string centralityLabel_;
+  std::string centralityMC_;
+
+  edm::InputTag centralityBinTag_;
+  edm::EDGetTokenT<int> centralityBinToken;
+  edm::Handle<int> cbin_;
```

Comment:
move edm::Handle s to the scope of methods that need them, unless there is very complex logic necessary to stash these as data members